[Merged by Bors] - chore(LinearAlgebra): split LinearIndependent into Defs, Basic, Lemmas

Vierkantor opened the pull request:
I was looking at reducing the imports and non-deffiness of `LinearAlgebra.Basis.Defs` and one big obstacle is that the basic dependencies in linear algebra already bring in a lot of theory. So I decided to try and split this large file. I ended up with three files:

* `LinearIndependent.Defs` contains the definitions and lemmas stating equivalent definitions to linear independence.
* `LinearIndependent.Basic` contains results provable without additional imports
* `LinearIndependent.Lemmas` contains the rest of the lemmas

---

- [x] depends on: #21886 
Jireh: not a strict dependency, but I think that PR should be merged first.

<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

Vierkantor committed:
mk_all!

Vierkantor committed:
Fix shake

j-loreaux commented:
This is going to conflict with #21886. Because that one is harder to review and to fix afterwards, I would suggest that this one waits for it instead. I'm marking as `blocked-by-other-PR` for that reason.

apnelson1 commented:
Is there a plan to get this finished soon? I'm champing at the bit to add more nice `LinearIndepOn` API, but it makes sense to wait for this.

Vierkantor commented:
Yes, sorry for the delay! I want to get this done today :)

Vierkantor committed:
chore(LinearAlgebra): split `LinearIndependent` into `Defs`, `Basic`, `Lemmas`

I was looking at reducing the imports and non-deffiness of `LinearAlgebra.Basis.Defs` and one big obstacle is that the basic dependencies in linear algebra already bring in a lot of theory. So I decided to try and split this large file. I ended up with three files:

* `LinearIndependent.Defs` contains the definitions and lemmas stating equivalent definitions to linear independence.
* `LinearIndependent.Basic` contains results provable without additional imports
* `LinearIndependent.Lemmas` contains the rest of the lemmas

Vierkantor committed:
Readd missing lemma.

Vierkantor commented:
Alright, should be ready for review again!

Vierkantor committed:
Shake

kim-em commented:
bors merge
